docs/content/en/docs/Design/do_transformation.md

```diff
@@ -0,0 +1,227 @@
+<!-- mdformat off(yaml frontmatter) -->
+---
+title: Data-oblivious Transformations
+weight: 9
+---
+<!-- mdformat on -->
+
+A data-oblivious program is one that decouples data input from program execution. Such programs exhibit control-flow and memory access patterns that are independent of their input(s). This programming model is necessary for expressing FHE programs. There are 3 major transformations applied to convert a conventional program into a data-oblivious program:
+
+### (1) If-Transformation
+
+If-operations conditioned on inputs create data-dependent control-flow in programs. `scf.if` operations should at least define a 'then' region (true path) and always terminate with `scf.yield` even when `scf.if` doesn't produce a result. To convert a data-dependent `scf.if` operation to an equivalent set of data-oblivious operations in MLIR, we hoist all safely speculatable operations in the `scf.if` operation and convert the `scf.yield` operation to an `arith.select` operation. The following code snippet demonstrates an application of this transformation.
+
+```mlir
+// Before applying If-transformation
+func.func @my_function(%input : i1) -> () {
```

Comment:
Given that the transformations now only apply to secret values, I think it's a good idea for the documentation to reflect that:
```suggestion
func.func @my_function(%input : i1 {secret.secret}) -> () {
```